### ISSUE TITLE
🔥 [firestore] update with FieldPath doesn't deep merge

### ISSUE THREAD

**Issue by iwikal** [invertase/react-native-firebase#2532]
## Issue

When specifying a field path with a firestore.FieldPath instead of a string, both `WriteBatch.update` and  `DocumentReference.update` fails to deep merge. For instance, given this document:
```js
firestore().doc('test/test').set({
  myMap: {
    initialKey: 'initialValue',
  },
  separateField: 'x',
})
```
...the following code...
```js
firestore().doc('test/test')
  .update('myMap.newKey', 'newValue')
```
...does exactly what I expect:
```json
{
  "myMap": {
    "initialKey": "initialValue",
    "newKey": "newValue"
  },
  "separateField": "x"
}
```

But this code does something unexpected!
```js
firestore()
  .doc('test/test')
  .update(new firestore.FieldPath('myMap', 'newKey'), 'newValue')
```
Result:
```json
{
  "myMap": {
    "newKey": "newValue"
  },
  "separateField": "x"
}
```
Note that the inital value in the map is gone, but the separate field remains. I tried the same thing with the firebase-admin sdk, and got the expected behaviour.

---

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [x] I'm not using Pods
- [ ] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [x] my application is an AndroidX application?
- [x] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [x] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
info Fetching system and libraries information...
System:
    OS: Linux 4.15 Ubuntu 18.04.3 LTS (Bionic Beaver)
    CPU: (4) x64 Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
    Memory: 2.61 GB / 15.55 GB
    Shell: 5.4.2 - /usr/bin/zsh
  Binaries:
    Node: 10.16.2 - ~/.nvm/versions/node/v10.16.2/bin/node
    Yarn: 1.17.3 - ~/.yarn/bin/yarn
    npm: 6.10.3 - ~/.nvm/versions/node/v10.16.2/bin/npm
  npmPackages:
    @react-native-community/cli: ^2.8.0 => 2.8.0 
    react: 16.8.6 => 16.8.6 
    react-native: 0.60.4 => 0.60.4

```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [x] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `5.5.6`
- **`Firebase` module(s) you're using that has the issue:**
  - `Firestore`
- **Are you using `TypeScript`?**
  - `Y`
  
</p>
</details>

---

Think `react-native-firebase` is great? Please consider supporting all of the project maintainers and contributors by donating via our [Open Collective](https://opencollective.com/react-native-firebase/donate) where all contributors can submit expenses. [[Learn More]](https://invertase.io/oss/react-native-firebase/contributing/donations-expenses)

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by Ehesp** [comment]
Interesting - thanks for the detailed report. I'll get this looked into.

**Comment by Ehesp** [comment]
Thanks for reporting. Turns out the fix was simple one! For some reason it was overcomplicating FieldPaths, trying to merge objects together when it can just pass the dot notated paths to native land.

**Comment by Salakar** [comment]
Hey 👋 this issue should now be fixed in v6.0.1, thanks. [[release notes]](https://invertase.io/oss/react-native-firebase/releases/v6.0.1)
